src/App.tsx

```diff
@@ -1,24 +1,184 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useEffect, useMemo, useRef, useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import * as todosService from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList/TodoList';
+import { TodoFooter } from './components/TodoFooter/TodoFooter';
+import { Status } from './types/Status';
+import { ErrorNotification } from './components/ErrorNotification';
 
-const USER_ID = 0;
+const USER_ID = 12042;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [status, setStatus] = useState<Status>(Status.all);
+  const [todoTitle, setTodoTitle] = useState<string>('');
+  const [loadingTodoIds, setLoadingTodoIds] = useState<number[]>([]);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+
+  const titleField = useRef<HTMLInputElement>(null);
+
+  const handleError = (errorMessage: string) => {
+    setError(errorMessage);
+    setTimeout(() => {
+      setError(null);
+    }, 3000);
+  };
+
+  useEffect(() => {
+    if (titleField.current) {
+      titleField.current.focus();
+    }
+
+    todosService.getTodos(USER_ID)
+      .then(setTodos)
+      .catch(() => (handleError('Unable to load todos')));
+  }, []);
+
+  const filteredTodos = useMemo(() => {
+    switch (status) {
+      case Status.all: {
+        return todos;
+      }
+
+      case Status.active: {
+        return todos.filter(todo => !todo.completed);
+      }
+
+      case Status.completed: {
+        return todos.filter(todo => todo.completed);
+      }
+
+      default: {
```

Comment:
```suggestion
      case Status.all:
      default: {
```
You can combine similar cases